compiler/src/iree/compiler/Codegen/Sandbox/LinalgTensorCodegenDriver.cpp

```diff
@@ -182,10 +183,16 @@ static SmallVector<int64_t> getVectorSizes(
   assert(canonicalVectorShape.size() >= linalgOp.getNumLoops() &&
          "Unexpected canonical vector shape or number of loops");
 
-  // Return the canonical vector shape subset based on the number of loops of
-  // the linalg op.
-  return {canonicalVectorShape.begin(),
-          std::next(canonicalVectorShape.begin(), linalgOp.getNumLoops())};
+  // Return the valid canonical vector shape subset based on the number of loops
+  // of the linalg op.
+  SmallVector<int64_t> vecSize(
+      canonicalVectorShape.take_front(linalgOp.getNumLoops()));
+  for (auto [idx, val] : llvm::enumerate(linalgOp.getStaticLoopRanges())) {
+    if (ShapedType::isDynamic(val)) continue;
+    vecSize[idx] = std::max(vecSize[idx], val);
+  }
+
+  return vecSize;
```

Comment:
@dcaballe can you take a look at this part since you have more context? Thank you!

Reply:
Hmmm... this doesn't look right. What is exactly the problem?